test.js

```diff
@@ -407,6 +407,29 @@ test('escapePath', t => {
 	});
 });
 
+test('deepKeys', t => {
+	t.deepEqual(deepKeys({
+		'a.b': {
+			c: {
+				d: [1, 2, 3],
+				e: '🦄',
+				f: 0,
+			},
+			'': {
+				a: 0,
+			},
+		},
+	}), [
+		'a\\.b.c.d',
```

Comment:
Shouldn’t this be
```suggestion
		'a\\.b.c.d.0',
		'a\\.b.c.d.1',
		'a\\.b.c.d.2',
```

Reply:
Arrays are intentionally not treated as objects to be traversed

Reply:
I see - sorry, I missed that in the code. What's the reason for that? It'd be a good idea to be deliberate about this because it could be considered a breaking change to respect arrays in future.

To give an example from the original motivating use-case, let's say we have some content in this shape:

```json5
{
  title: 'FAQ page',
  blurb: 'Questions and answers',
  questions: [
    { question: 'What is ABC?', answer: 'ABC is DEF' },
    { question: 'What is XYZ?', answer: 'XYZ is 123' },
  ]
}
```

If we wanted to get some key-value pairs we'd want to send to the translation service would be:

```json5
[
  ['title', 'FAQ page'],
  ['blurb', 'Questions and answers'],
  ['questions[0].question', 'What is ABC?'],
  ['questions[0].answer', 'ABC is DEF'],
  ['questions[1].question', 'What is XYZ?'],
  ['questions[1].answer', 'XYZ is 123'],
]
```

If `deepKeys` returns `['title', 'blurb', 'questions']` - then it can't really be used for this because users would have to write their own recursive function to handle arrays.